Conductor/Client/ApiClient.cs

```diff
@@ -165,44 +168,49 @@ private byte[] GetBytes(Stream input)
             Dictionary<String, FileParameter> fileParams, Dictionary<String, String> pathParams,
             String contentType, Configuration configuration = null)
+        {
+            int retryCount = 0;
+            RestResponse response = RetryRestClientCallApi(path, method, queryParams, postBody, headerParams,
+                formParams, fileParams, pathParams, contentType, configuration, ref retryCount);
+
+            return (Object)response;
+        }
+
+        private RestResponse RetryRestClientCallApi(String path, Method method, List<KeyValuePair<String, String>> queryParams, Object postBody,
+            Dictionary<String, String> headerParams, Dictionary<String, String> formParams,
+            Dictionary<String, FileParameter> fileParams, Dictionary<String, String> pathParams,
+            String contentType, Configuration configuration, ref int retryCount)
         {
             RestResponse response = null;
-            bool isTokenRefreshed = false;
-            do
+            while (retryCount < Constants.MAX_TOKEN_REFRESH_RETRY_COUNT)
             {
+                bool isTokenRefreshed = false; // reset refreshStatusflag
+
                 var request = PrepareRequest(
                     path, method, queryParams, postBody, headerParams, formParams, fileParams,
                     pathParams, contentType);
 
-                response = ExecuteRestClientAPI(request, method);
+                InterceptRequest(request);
+                response = RestClient.Execute(request, method);
+                InterceptResponse(request, response);
+                FormatHeaders(response);
 
                 if (response.StatusCode == System.Net.HttpStatusCode.Unauthorized)
                 {
                     var JsonContent = JsonConvert.DeserializeObject<JObject>(response.Content);
 
-                    if (JsonContent["error"].ToString() == "EXPIRED_TOKEN" && !isTokenRefreshed)
+                    if (JsonContent["error"].ToString() == "EXPIRED_TOKEN" && retryCount < Constants.MAX_TOKEN_REFRESH_RETRY_COUNT)
                     {
                         string refreshToken = configuration.GetRefreshToken();
                         headerParams["X-Authorization"] = refreshToken;
                         isTokenRefreshed = true;
+                        retryCount++;
                     }
-                    else
-                        isTokenRefreshed = false;
                 }
-                else
-                    isTokenRefreshed = false;
-            } while (isTokenRefreshed);
-
-            return (Object)response;
-        }
-
-        private RestResponse ExecuteRestClientAPI(RestRequest request, Method method)
-        {
-            InterceptRequest(request);
-            var response = RestClient.Execute(request, method);
-            InterceptResponse(request, response);
-            FormatHeaders(response);
 
+                if (!isTokenRefreshed)
```

Comment:
Do we need retryCount and isTokenRefreshed both? I think retryCount is acting as a circuit breaker and once the API response is correct we should just set the incoming token and break the loop.

Reply:
> Do we need retryCount and isTokenRefreshed both? I think retryCount is acting as a circuit breaker and once the API response is correct we should just set the incoming token and break the loop.

@manan164 Makes sense. Modified it, please take a look. Thanks